pandas/_libs/lib.pyx

```diff
@@ -221,7 +221,7 @@ def is_iterator(obj: object) -> bool:
     return PyIter_Check(obj)
 
 
-def item_from_zerodim(val: object) -> object:
+cpdef object item_from_zerodim(object val):
```

Comment:
can you add inline here?

Reply:
could i guess, i dont think it makes a difference

Reply:
on balance i prefer not to mix "inline" with "cpdef"

Reply:
right that’s the question does this make a perf difference? as u have it vs master

Reply:
huh, if map_infer is somewhat slower in this PR than in master, and its _worse_ with `inline`.  so im going to revert this edit, remove the TODO comments, and in a separate branch look into if we can improve item_from_zerodim

Reply:
updated+green

Reply:
I can open an issue about this separately, but our current inline strategy _may_ be doing more harm than good. GCC has a `-Winline` flag that will tell you when a function declared with inline isn't actually inlined, and running this on Linux appears to produce 11 MB worth of warnings with a clean build of pandas

Reply:
Yikes.  Addressing the larger issue separately makes sense.  For the purposes of this PR, is there something that should be changed?

Reply:
Things lgtm here as is, but let's see what Jeff thinks

Reply:
raised issue in #33926

Reply:
no this is fine, but yeah looking at this generally is prob a good idea.